src/stores/modalContext.test.tsx

```diff
@@ -0,0 +1,366 @@
+        })
+
+        updateWidget({
+          _id: mockCollectionId,
+          title: 'Test Collection',
+          type: 'Collection',
+        })
+
+        updateCustomLinkLabel('My Custom Label', false, true)
+
+        modalRef?.current?.toggleModal(undefined, true)
+      }
+
+      return (
+        <div>
+          <div id="modal-root" className="sfds" />
+          <button type="button" onClick={setupFunc}>
+            Add link
+          </button>
+          <CustomModal />
+        </div>
+      )
+    }
+
+    renderWithAuthAndApollo(
+      <ModalProvider>
+        <TestComponent />
+      </ModalProvider>,
+      {},
+      addBookmarkMock
+    )
+
+    const openModalButton = screen.getByRole('button', {
+      name: 'Add link',
+    })
+    expect(openModalButton).toBeInTheDocument()
+
+    await user.click(openModalButton)
+
+    // Add in link details
+    const urlInput = screen.getByLabelText('URL')
+
+    await user.clear(urlInput)
+    await user.type(urlInput, 'example.com')
+
+    // Save link
+    const saveButton = screen.getByText('Save custom link')
+    expect(saveButton).toBeInTheDocument()
+
+    await user.click(saveButton)
+
+    // Again, not really sure what the assertion should be here
+    // expect(addBookmarkMock[0].result.data.label).toEqual('fail')
+  })
+
+  it('tests editing a bookmark', async () => {
+    const user = userEvent.setup()
+
+    const TestComponent = () => {
+      const {
+        modalRef,
+        updateModalId,
+        updateModalText,
+        updateWidget,
+        updateBookmark,
+      } = useModalContext()
+
+      const setupFunc = () => {
+        updateModalId('editCustomLinkModal')
+        updateModalText({
+          headingText: 'Edit custom link',
+        })
+
+        updateWidget({
+          _id: mockCollectionIdForEditBookmark,
+          title: 'Test Collection',
+          type: 'Collection',
+        })
+
+        updateBookmark(mockBookmark)
+
+        modalRef?.current?.toggleModal(undefined, true)
+      }
+
+      return (
+        <div>
+          <div id="modal-root" className="sfds" />
+          <button type="button" onClick={setupFunc}>
+            Edit link
+          </button>
+          <CustomModal />
+        </div>
+      )
+    }
+
+    renderWithAuthAndApollo(
+      <ModalProvider>
+        <TestComponent />
+      </ModalProvider>,
+      {},
+      editBookmarkMock
+    )
+
+    // Open modal
+    const openModalButton = screen.getByRole('button', {
+      name: 'Edit link',
+    })
+    expect(openModalButton).toBeInTheDocument()
+
+    await user.click(openModalButton)
+
+    // Update label and save
+    const nameInput = screen.getByLabelText('Name')
+    await user.clear(nameInput)
+    await user.type(nameInput, 'Custom Label')
+
+    const saveButton = screen.getByText('Save custom link')
+    expect(saveButton).toBeInTheDocument()
+
+    await user.click(saveButton)
+
+    // Need an assertion?
```

Comment:
Question: Can you add another expect here? Maybe to verify the custom link text is displayed? Would you like to pair on these?